bgpd/bgp_fsm.c

```diff
@@ -935,9 +936,28 @@ static void bgp_update_delay_process_status_change(struct peer *peer)
    read/write and timer thread. */
 void bgp_fsm_change_status(struct peer *peer, int status)
 {
+	struct bgp *bgp;
 
 	bgp_dump_state(peer, peer->status, status);
 
+	bgp = peer->bgp;
+	if (bgp) {
```

Comment:
Can you explain to me how peer->bgp is null?  This is causing the new SA warnings.  I do not see a code path where we have a peer data structure and bgp is NULL.